### PR TITLE
Add ExpectNotSendToDestination ExpectInvocation

### LINKED ISSUE #84: There is no ExpectNotSendToDestination
I came across a case where I needed this and it was not in the testing framework

### PR DESCRIPTION
**Pull request by feinoujc**
closes #84

### CODE REVIEW

File: src/NServiceBus.Testing/Handler.cs [hunk 1]
```diff
@@ -252,6 +252,15 @@ public Handler<T> ExpectSendToDestination<TMessage>(Func<TMessage, string, bool>
         }
 
         /// <summary>
+        /// Check that the handler does not send the given message type to the given destination.
+        /// </summary>
+        public Handler<T> ExpectNotSendToDestination<TMessage>(Func<TMessage, string, bool> check)
```
Comment (timbussmann) [inline src/NServiceBus.Testing/Handler.cs]: I think that parameter could be optional

**Comment by timbussmann** [comment]
Hey @feinoujc 

The changes look very good to me :+1: 
The build server fails because the public API has changed by adding a new assertion. You need to manually approve the API which you can do by running the `ApiApproval` unit test. If run in debug, it will copy a command to approve the API into your clipboard. Open a console, paste the command, execute it and push the changed approved.txt file which will make the buildserver happy.

Of course I can do that for you, which requires me to pull over your branch into our repository (and therefore opening a new PR) if you prefer that. Just let me know.

**Comment by timbussmann** [comment]
ping @Particular/nservicebus-maintainers

**Comment by feinoujc** [comment]
@timbussmann no problem. I've made these changes

**Comment by timbussmann** [comment]
@feinoujc thanks a lot for the PR to provide the missing assertion, awesome work! :tada: 

I'll cherry pick the changes into the current release branch to make sure it's included in the 6.0 release.

**Comment by danielmarbach** [comment]
Let's ping malki for a little gift

**Comment by malkinservicebus** [comment]
Hi @feinoujc 

Thank you for your contribution to NServiceBus code! 

We would like to send you a token of appreciation. Please forward your postal address to  operations@particular.net  so that we can send you a special gift.

**Pull request merged**